### PR TITLE
[Snappi] Added multi dut ecn red accuracy test case

### PR DESCRIPTION
**Pull request by vikumarks**
### Description of PR
- Test to determine the RED/ECN accuracy of marking packets at the egress (a.k.a dequeue ECN) 

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
To verify the accuracy of ECN marking in multi-dut/multi-asic scenario. 
#### How did you do it?
The existing RDMA cases for single dut has been modified to accommodate T2 topology, where separate fixtures has been added that takes into account the chassis type and asic choice as user input, so that the script supports both T1 and T2 topology
#### How did you verify/test it?
Tested against voq chassis with SONiC installed in it 
#### Any platform specific information?
Tested against J2C+ 
#### Supported testbed topology if it's a new test case?
T2, T1 & T0 
### Documentation
https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testplan/ecn/ECN_ACCURACY_README.md

### Output
`============================= test session starts ==============================
platform linux2 -- Python 2.7.18, pytest-4.6.11, py-1.11.0, pluggy-0.13.1
ansible: 2.8.20
rootdir: /var/AzDevOps/sonic-mgmt/tests, inifile: pytest.ini
plugins: allure-pytest-2.8.22, celery-4.4.7, ansible-2.2.4, forked-1.3.0, html-1.22.1, metadata-1.11.0, repeat-0.9.1, xdist-1.28.0

----------------------------- live log collection ------------------------------
18:22:20 __init__.pytest_collection_modifyitems   L0582 INFO   | Available basic facts that can be used in conditional skip:
{
  "topo_type": "ptf", 
  "testbed": "vms-snappi-sonic", 
  "topo_name": "ptf64"
}

-------------------------------- live log setup --------------------------------
18:22:20 __init__.set_default                     L0054 INFO   | Completeness level not set during test execution. Setting to default level: CompletenessLevel.basic
18:22:20 __init__.check_test_completeness         L0152 INFO   | Test has no defined levels. Continue without test completeness checks
18:22:27 ptfhost_utils.run_icmp_responder_session L0239 INFO   | Skip running icmp_responder at session level, it is only for dualtor testbed with active-active mux ports.
18:22:27 conftest.creds_on_dut                    L0704 INFO   | dut sonic-s6100-dut belongs to groups [u'snappi-sonic', u'sonic', u'sonic_dell64_40', 'fanout']
18:22:27 conftest.creds_on_dut                    L0728 INFO   | skip empty var file ../ansible/group_vars/all/corefile_uploader.yml
18:22:27 conftest.creds_on_dut                    L0728 INFO   | skip empty var file ../ansible/group_vars/all/env.yml
18:22:27 conftest.nbrhosts                        L0521 INFO   | No VMs exist for this topology: ptf64
18:22:27 conftest.core_dump_and_config_check      L2032 INFO   | Collecting core dumps before test on sonic-s6100-dut
18:22:28 conftest.core_dump_and_config_check      L2041 INFO   | Collecting running config before test on sonic-s6100-dut
18:22:32 __init__.sanity_check                    L0125 INFO   | Skip sanity check according to command line argument
18:22:32 conftest.generate_params_dut_hostname    L1108 INFO   | Using DUTs ['sonic-s6100-dut'] in testbed 'vms-snappi-sonic'
18:22:32 conftest.rand_one_dut_hostname           L0378 INFO   | Randomly select dut sonic-s6100-dut for testing
18:22:32 __init__._fixture_generator_decorator    L0081 INFO   | -------------------- fixture enable_packet_aging_after_test setup starts --------------------
18:22:32 __init__._fixture_generator_decorator    L0085 INFO   | -------------------- fixture enable_packet_aging_after_test setup ends --------------------
18:22:32 __init__._fixture_generator_decorator    L0081 INFO   | -------------------- fixture rand_lossless_prio setup starts --------------------
18:22:32 __init__._fixture_generator_decorator    L0085 INFO   | -------------------- fixture rand_lossless_prio setup ends --------------------
18:22:32 __init__._fixture_generator_decorator    L0081 INFO   | -------------------- fixture rand_lossy_prio setup starts --------------------
18:22:32 __init__._fixture_generator_decorator    L0085 INFO   | -------------------- fixture rand_lossy_prio setup ends --------------------
18:22:32 __init__._fixture_generator_decorator    L0081 INFO   | -------------------- fixture start_pfcwd_after_test setup starts --------------------
18:22:32 __init__._fixture_generator_decorator    L0085 INFO   | -------------------- fixture start_pfcwd_after_test setup ends --------------------
18:22:32 __init__._fixture_func_decorator         L0069 INFO   | -------------------- fixture snappi_api_serv_ip setup starts --------------------
18:22:32 __init__._fixture_func_decorator         L0076 INFO   | -------------------- fixture snappi_api_serv_ip setup ends --------------------
18:22:32 __init__._fixture_func_decorator         L0069 INFO   | -------------------- fixture snappi_api_serv_port setup starts --------------------
18:22:33 __init__._fixture_func_decorator         L0076 INFO   | -------------------- fixture snappi_api_serv_port setup ends --------------------
18:22:33 __init__._fixture_generator_decorator    L0081 INFO   | -------------------- fixture snappi_api setup starts --------------------
18:22:33 __init__._fixture_generator_decorator    L0085 INFO   | -------------------- fixture snappi_api setup ends --------------------
18:22:33 __init__._fixture_func_decorator         L0069 INFO   | -------------------- fixture prio_dscp_map setup starts --------------------
18:22:34 __init__._fixture_func_decorator         L0076 INFO   | -------------------- fixture prio_dscp_map setup ends --------------------
18:22:34 __init__.loganalyzer                     L0045 INFO   | Log analyzer is disabled
18:22:34 __init__._fixture_func_decorator         L0069 INFO   | -------------------- fixture get_multidut_snappi_ports setup starts --------------------
18:22:34 __init__._fixture_func_decorator         L0076 INFO   | -------------------- fixture get_multidut_snappi_ports setup ends --------------------
-------------------------------- live log call ---------------------------------
18:22:34 snappi_fixtures.__intf_config_multidut   L0789 INFO   | Configuring Dut: sonic-s6100-dut with port Ethernet4 with IP 20.1.1.1/8
18:22:36 snappi_fixtures.__intf_config_multidut   L0789 INFO   | Configuring Dut: sonic-s6100-dut with port Ethernet0 with IP 21.1.1.1/8
18:22:37 test_multidut_red_accuracy_with_snappi.t L0104 INFO   | Running ECN red accuracy test with ECN params: {'kmax': 2000000, 'kmin': 500000, 'pmax': 5}
18:22:37 test_multidut_red_accuracy_with_snappi.t L0105 INFO   | Running ECN red accuracy test for 1 iterations
18:22:37 multidut_helper.run_ecn_test             L0067 INFO   | Stopping PFC watchdog
18:22:40 multidut_helper.run_ecn_test             L0070 INFO   | Disabling packet aging if necessary
18:22:42 multidut_helper.run_ecn_test             L0075 INFO   | Configuring WRED and ECN thresholds
18:22:48 multidut_helper.run_ecn_test             L0088 INFO   | Enabling ECN markings
18:22:57 multidut_helper.run_ecn_test             L0119 INFO   | Generating base flow config
18:22:57 multidut_helper.run_ecn_test             L0124 INFO   | Setting test flow config params
18:22:57 multidut_helper.run_ecn_test             L0132 INFO   | Setting pause flow config params
18:22:57 multidut_helper.run_ecn_test             L0146 INFO   | Generating test flows
18:22:57 multidut_helper.run_ecn_test             L0152 INFO   | Generating pause flows
18:22:57 multidut_helper.run_ecn_test             L0163 INFO   | Setting packet capture port to Port 0
18:22:57 multidut_helper.run_ecn_test             L0167 INFO   | Running 1 iteration(s)
18:22:57 multidut_helper.run_ecn_test             L0169 INFO   | Running iteration 0
18:22:57 multidut_helper.run_ecn_test             L0171 INFO   | Packet capture file: ECN_cap-0.pcapng
18:22:57 multidut_helper.run_ecn_test             L0178 INFO   | Running traffic
18:22:57 connection._warn                         L0246 WARNING| Verification of certificates is disabled
18:22:57 connection._info                         L0243 INFO   | Determining the platform and rest_port using the 10.36.78.135 address...
18:22:57 connection._warn                         L0246 WARNING| Unable to connect to http://10.36.78.135:443.
18:22:57 connection._info                         L0243 INFO   | Connection established to `https://10.36.78.135:443 on linux`
18:23:11 connection._info                         L0243 INFO   | Using IxNetwork api server version 9.30.2212.7
18:23:11 connection._info                         L0243 INFO   | User info IxNetwork/ixnetworkweb/admin-118-24934
18:23:11 snappi_api.info                          L1132 INFO   | snappi-0.9.1
18:23:11 snappi_api.info                          L1132 INFO   | snappi_ixnetwork-0.9.1
18:23:11 snappi_api.info                          L1132 INFO   | ixnetwork_restpy-1.0.64
18:23:12 snappi_api.info                          L1132 INFO   | Config validation 0.005s
18:23:15 snappi_api.info                          L1132 INFO   | Ports configuration 3.037s
18:23:16 snappi_api.info                          L1132 INFO   | Captures configuration 0.496s
18:23:18 snappi_api.info                          L1132 INFO   | Add location hosts [10.36.78.53] 2.308s
18:23:23 snappi_api.info                          L1132 INFO   | Location hosts ready [10.36.78.53] 4.176s
18:23:23 snappi_api.info                          L1132 INFO   | Speed conversion is not require for (port.name, speed) : [('Port 0', u'novusHundredGigNonFanOut'), ('Port 1', u'novusHundredGigNonFanOut')]
18:23:23 snappi_api.info                          L1132 INFO   | Aggregation mode speed change 0.449s
18:23:28 snappi_api.info                          L1132 INFO   | Location preemption [10.36.78.53;6;2, 10.36.78.53;6;1] 0.117s
18:23:52 snappi_api.info                          L1132 INFO   | Location connect [Port 0, Port 1] 24.515s
18:23:52 snappi_api.warning                       L1138 WARNING| Port 0 connectedLinkDown
18:23:52 snappi_api.warning                       L1138 WARNING| Port 1 connectedLinkDown
18:23:52 snappi_api.info                          L1132 INFO   | Location state check [Port 0, Port 1] 0.270s
18:23:52 snappi_api.info                          L1132 INFO   | Location configuration 36.640s
18:24:05 snappi_api.info                          L1132 INFO   | Layer1 configuration 12.249s
18:24:05 snappi_api.info                          L1132 INFO   | Lag Configuration 0.089s
18:24:05 snappi_api.info                          L1132 INFO   | Convert device config : 0.249s
18:24:05 snappi_api.info                          L1132 INFO   | Create IxNetwork device config : 0.000s
18:24:06 snappi_api.info                          L1132 INFO   | Push IxNetwork device config : 0.584s
18:24:06 snappi_api.info                          L1132 INFO   | Devices configuration 0.912s
18:24:08 snappi_api.info                          L1132 INFO   | Flows configuration 2.554s
18:24:15 snappi_api.info                          L1132 INFO   | Start interfaces 7.030s
18:24:16 snappi_api.info                          L1132 INFO   | IxNet - The Traffic Item was modified. Please perform a Traffic Generate to update the associated traffic Flow Groups
18:24:16 traffic_generation.run_traffic           L0549 INFO   | Wait for Arp to Resolve ...
18:24:17 traffic_generation.run_traffic           L0559 INFO   | Starting packet capture ...
18:24:20 snappi_api.info                          L1132 INFO   | Captures start 2.214s
18:24:20 traffic_generation.run_traffic           L0567 INFO   | Starting transmit on all flows ...
18:24:23 snappi_api.info                          L1132 INFO   | Flows generate/apply 2.989s
18:24:35 snappi_api.info                          L1132 INFO   | Flows clear statistics 12.025s
18:24:37 snappi_api.info                          L1132 INFO   | Captures start 1.823s
18:24:44 snappi_api.info                          L1132 INFO   | Flows start 6.562s
18:24:47 traffic_generation.run_traffic           L0608 INFO   | All test and background traffic flows stopped
18:24:49 traffic_generation.run_traffic           L0619 INFO   | Stopping packet capture ...
18:24:50 snappi_api.info                          L1132 INFO   | Captures stop 1.269s
18:24:50 traffic_generation.run_traffic           L0625 INFO   | Retrieving and saving packet capture to ECN_cap-0.pcapng
18:27:25 snappi_api.warning                       L1138 WARNING| Capture was not stopped for this port Port 0
18:27:25 snappi_api.info                          L1132 INFO   | Captures stop 154.431s
18:27:27 traffic_generation.run_traffic           L0631 INFO   | Dumping per-flow statistics
18:27:28 traffic_generation.run_traffic           L0635 INFO   | Stopping transmit on all remaining flows
18:27:29 read_pcap.get_ip_pkts                    L0071 INFO   | Reading packets from pcap file -> ECN_cap-0.pcapng
18:27:29 read_pcap.get_ip_pkts                    L0072 INFO   | Extracting ethernet frames from pcap file
18:27:29 test_multidut_red_accuracy_with_snappi.t L0127 INFO   | Instantiating a queue length vs. ECN marking probability dictionary
18:27:29 test_multidut_red_accuracy_with_snappi.t L0133 INFO   | Check that all packets are captured for each iteration
18:27:29 read_pcap.is_ecn_marked                  L0093 INFO   | Checking if the packet is ECN congestion marked
18:27:29 read_pcap.is_ecn_marked                  L0093 INFO   | Checking if the packet is ECN congestion marked
18:27:29 test_multidut_red_accuracy_with_snappi.t L0148 INFO   | ------- Dumping queue length vs. ECN marking probability data ------
18:27:29 test_multidut_red_accuracy_with_snappi.t L0156 INFO   | Reloading config to teardown ECN config
18:27:29 config_reload.config_reload              L0093 INFO   | reloading config_db
18:28:25 processes_utils.wait_critical_processes  L0071 INFO   | Wait until all critical processes are healthy in 300 sec
18:28:25 processes_utils._all_critical_processes_ L0039 INFO   | Check critical processes status
18:29:02 processes_utils._all_critical_processes_ L0039 INFO   | Check critical processes status
18:29:11 config_reload.config_reload              L0093 INFO   | reloading config_db
18:30:16 processes_utils.wait_critical_processes  L0071 INFO   | Wait until all critical processes are healthy in 300 sec
18:30:16 processes_utils._all_critical_processes_ L0039 INFO   | Check critical processes status
PASSED                                                                   [100%]
------------------------------ live log teardown -------------------------------
18:30:30 __init__._fixture_generator_decorator    L0093 INFO   | -------------------- fixture snappi_api teardown starts --------------------
18:31:02 __init__._fixture_generator_decorator    L0102 INFO   | -------------------- fixture snappi_api teardown ends --------------------
18:31:02 __init__._fixture_generator_decorator    L0093 INFO   | -------------------- fixture start_pfcwd_after_test teardown starts --------------------
18:31:04 __init__._fixture_generator_decorator    L0102 INFO   | -------------------- fixture start_pfcwd_after_test teardown ends --------------------
18:31:04 __init__._fixture_generator_decorator    L0093 INFO   | -------------------- fixture rand_lossy_prio teardown starts --------------------
18:31:04 __init__._fixture_generator_decorator    L0102 INFO   | -------------------- fixture rand_lossy_prio teardown ends --------------------
18:31:04 __init__._fixture_generator_decorator    L0093 INFO   | -------------------- fixture rand_lossless_prio teardown starts --------------------
18:31:04 __init__._fixture_generator_decorator    L0102 INFO   | -------------------- fixture rand_lossless_prio teardown ends --------------------
18:31:04 __init__._fixture_generator_decorator    L0093 INFO   | -------------------- fixture enable_packet_aging_after_test teardown starts --------------------
18:31:05 __init__._fixture_generator_decorator    L0102 INFO   | -------------------- fixture enable_packet_aging_after_test teardown ends --------------------
18:31:05 conftest.core_dump_and_config_check      L2069 INFO   | Collecting core dumps after test on sonic-s6100-dut
18:31:06 conftest.core_dump_and_config_check      L2086 INFO   | Collecting running config after test on sonic-s6100-dut
18:31:07 conftest.core_dump_and_config_check      L2211 WARNING| Core dump or config check failed for test_multidut_red_accuracy_with_snappi.py, results: {"core_dump_check": {"new_core_dumps": {"sonic-s6100-dut": []}, "pass": true}, "config_db_check": {"cur_only_config": {"sonic-s6100-dut": {"null": {"INTERFACE": {"Ethernet8": {}, "Ethernet0|2000:1::1/16": {}, "Ethernet0": {}, "Ethernet4": {}, "Ethernet12|23.1.1.1/8": {}, "Ethernet12|2003:1::1/16": {}, "Ethernet8|2002:1::1/16": {}, "Ethernet4|2001:1::1/16": {}, "Ethernet8|22.1.1.1/8": {}, "Ethernet4|21.1.1.1/8": {}, "Ethernet0|20.1.1.1/8": {}, "Ethernet12": {}}}}}, "inconsistent_config": {"sonic-s6100-dut": {"null": {}}}, "pre_only_config": {"sonic-s6100-dut": {"null": {"VLAN_MEMBER": {"Vlan2|Ethernet8": {"tagging_mode": "untagged"}, "Vlan2|Ethernet0": {"tagging_mode": "untagged"}, "Vlan2|Ethernet12": {"tagging_mode": "untagged"}, "Vlan2|Ethernet4": {"tagging_mode": "untagged"}}}}}, "pass": false}}

=============================== warnings summary ===============================
/usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:545
  /usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:545: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: tests.common.plugins.loganalyzer
    self.import_plugin(import_spec)

/usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:545
  /usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:545: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: tests.common.plugins.sanity_check
    self.import_plugin(import_spec)

/usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:545
  /usr/local/lib/python2.7/dist-packages/_pytest/config/__init__.py:545: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: tests.common.dualtor
    self.import_plugin(import_spec)

snappi_tests/dinesh_response_time_pr/test_multidut_red_accuracy_with_snappi.py::test_red_accuracy[linecard_configuration_set0-non_chassis_single_line_card]
  /usr/local/lib/python2.7/dist-packages/pytest_ansible/module_dispatcher/v28.py:81: UserWarning: provided hosts list is empty, only localhost is available
    warnings.warn("provided hosts list is empty, only localhost is available")

-- Docs: https://docs.pytest.org/en/latest/warnings.html
---------------------------- live log sessionfinish ----------------------------
18:31:07 __init__.pytest_terminal_summary         L0064 INFO   | Can not get Allure report URL. Please check logs
==================== 1 passed, 4 warnings in 528.54 seconds ====================`

### CODE REVIEW

**Comment by kamalsahu0001** [comment]
Testplan for ECN red accuracy is same for both singledut and multidut
https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testplan/ecn/ECN_ACCURACY_README.md

**Comment by dks0692** [comment]
@developfast Can you please review

**Review by vmittal-msft — APPROVED** [review]
This seems to be dependent on https://github.com/sonic-net/sonic-mgmt/pull/7914. Also, could we verify on single dut-single asic, single dut-multi asic, multi dut ? Please paste results here. Also, when the test is run, is it going to verify all these combinations or going to randonly pick one out of 3 ?

**Comment by vmittal-msft** [comment]
@vikumarks please fix failures. 
@yxieca @wangxin please help merge this.

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        pytest_require(False, "Need Minimum of 2 ports for the test")
+
+    snappi_ports = get_multidut_tgen_peer_port_set(line_card_choice, snappi_port_list, config_set, 2)
+
+    testbed_config, port_config_list, snappi_ports = snappi_dut_base_config(dut_list,
+                                                                            snappi_ports,
+                                                                            snappi_api)
+
+    _, lossless_prio = rand_one_dut_lossless_prio.split('|')
+    skip_ecn_tests(duthost1)
```
Comment (developfast) [inline tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py:83]: put maybe on the same line as a `or` case. This test cannot possibly run if either doesn't meet the criteria.

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            if is_ecn_marked(ip_pkt):
+                queue_mark_cnt[queue_len] += 1
+
+    # Dump queue length vs. ECN marking probability into logger file """
+    logger.info("------- Dumping queue length vs. ECN marking probability data ------")
+    output_table = []
+    queue_mark_cnt = collections.OrderedDict(sorted(queue_mark_cnt.items()))
+    for queue, mark_cnt in list(queue_mark_cnt.items()):
+        output_table.append([queue, float(mark_cnt)/num_iterations])
+    # logger.info(tabulate(output_table, headers=['Queue Length', 'ECN Marking Probability']))
```
Comment (developfast) [inline tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py:148]: remove this comment or log it?

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                      'Only capture {}/{} packets in round {}'.format(len(ip_pkts), data_flow_pkt_count, i))
+
+        for j in range(data_flow_pkt_count):
+            ip_pkt = ip_pkts[j]
+            queue_len = (data_flow_pkt_count - j) * data_flow_pkt_size
+
+            if is_ecn_marked(ip_pkt):
+                queue_mark_cnt[queue_len] += 1
+
+    # Dump queue length vs. ECN marking probability into logger file """
```
Comment (developfast) [inline tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py:142]: Remove the """ at the end

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    logger.info("Instantiating a queue length vs. ECN marking probability dictionary")
+    queue_mark_cnt = {}
+    for i in range(data_flow_pkt_count):
+        queue_len = (data_flow_pkt_count - i) * data_flow_pkt_size
+        queue_mark_cnt[queue_len] = 0
+
+    logger.info("Check that all packets are captured for each iteration")
+    for i in range(num_iterations):
+        ip_pkts = ip_pkts_list[i]
+        # Check if we capture all the packets in each round """
```
Comment (developfast) [inline tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py:131]: Remove the """ at the end

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                                port_config_list=port_config_list,
+                                conn_data=conn_graph_facts,
+                                fanout_data=fanout_graph_facts,
+                                dut_port=snappi_ports[0]['peer_port'],
+                                lossless_prio=lossless_prio,
+                                prio_dscp_map=prio_dscp_map,
+                                iters=num_iterations,
+                                snappi_extra_params=snappi_extra_params)
+
+    # Check if we capture packets of all the rounds """
```
Comment (developfast) [inline tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py:118]: Remove the """ at the end

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py (hunk 2 above)
Comment (developfast) [reply tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py:148]: Recommend logging it

**Review by developfast — CHANGES_REQUESTED** [review]
as comments

**Comment by developfast** [comment]
@vikumarks can you pls check the test error?

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py (hunk 5 above)
Comment (vikumarks) [reply tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py]: Fixed

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py (hunk 4 above)
Comment (vikumarks) [reply tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py]: Fixed

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py (hunk 3 above)
Comment (vikumarks) [reply tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py]: Fixed

File: tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py (hunk 2 above)
Comment (vikumarks) [reply tests/snappi_tests/multidut/ecn/test_multidut_red_accuracy_with_snappi.py]: Fixed

**Comment by vikumarks** [comment]
> @vikumarks can you pls check the test error?

@developfast This PR depends on PR7914  for multidut_helper.py file.

**Comment by vikumarks** [comment]
@developfast can we merge this PR

**Comment by vmittal-msft** [comment]
@vikumarks it is failing in test. please check.

@yxieca @developfast @wangxin please help merge.

**Comment by vikumarks** [comment]
> @vikumarks it is failing in test. please check.
> 
> @yxieca @developfast @wangxin please help merge.

@vmittal-msft All checks have passed 
@yxieca @developfast @wangxin  please help merge

**Comment by vmittal-msft** [comment]
@yxieca @wangxin please help

**Review by developfast — APPROVED** [review]
lgtm

**Pull request merged**